web-server-lib/web-server/http/request.rkt

```diff
@@ -140,8 +135,23 @@
 ;; **************************************************
 ;; complete-request
 
-(define (hex-string->number s)
-  (string->number s 16))
+(define (bytes->nonnegative-integer bs [base 10])
+  (and (not (zero? (bytes-length bs)))
+       (case base
+         [(10)
+          (for/fold ([n 0])
+                    ([b (in-bytes bs)] #:when n)
+            (and (>= b 48) (<= b 58) (+ (* n 10) (- b 48))))]
+         [(16)
+          (for/fold ([n 0])
+                    ([b (in-bytes bs)] #:when n)
+            (or
+             (and (>= b 48) (<= b 57)  (+ (* n 16) (- b 48))) ;; 0-9
+             (and (>= b 65) (<= b 70)  (+ (* n 16) (- b 55))) ;; A-F
+             (and (>= b 97) (<= b 102) (+ (* n 16) (- b 87))) ;; a-f
+             ))]
+         [else
+          (raise-argument-error 'bytes->number "(or/c 10 16)" base)])))
```

Comment:
This change seems like something that should be integrated into Racket core

Reply:
(or at least a utility library)

Reply:
Seems similar to integer-bytes->integer.

Reply:
The closest thing in core to this function is `string->number`. `integer-bytes->integer` is meant for interpreting a series of bytes as an integer, so it wouldn't be appropriate here:

```racket
> (bytes->nonnegative-integer #"123")
123
> (integer-bytes->integer #"123" #t)
integer-bytes->integer: length is not 1, 2, 4, or 8 bytes
  length: 3
 [,bt for context]
> (integer-bytes->integer #"1234" #t)
875770417
> (integer-bytes->integer #"\xFF\x00\x00\x00" #t #f)
255
```

I agree that it would be nice to have a `bytes->number` in core, though. I'll see if I can find time to do that after this PR. This specific version, though, is specialized to what the web server needs (doesn't support negative numbers, floats), so it's more work than just copying it over.

Reply:
Ok